app/components/Container.jsx

```diff
@@ -1,16 +1,61 @@
 import React from 'react';
-import { RouteHandler } from 'react-router';
+import {RouteHandler} from 'react-router';
 import ContainerNav from './ContainerNav';
+import {onSitesWithinRadius, getLocation} from '../utils/geo';
+import {getSiteByKey} from '../utils/sites';
 
 // styles
 import '../styles/components/Container';
 
 export default class Home extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      origin: new google.maps.LatLng(34.04935261524454, -118.24610710144043),
+      sites: [],
+    };
+    this.getSites = this.getSites.bind(this);
+    this.onMarkerClick = this.onMarkerClick.bind(this);
+    this.handleCloseClick = this.handleCloseClick.bind(this);
+  }
+  componentDidMount() {
+    getLocation().then(this.getSites);
+  }
+  onMarkerClick(currentMarker) {
+    const sites = this.state.sites;
+    const index = sites.indexOf(currentMarker);
+    const site = sites[index];
+    site.showInfo = true;
```

Comment:
Let's refactor this to get around mutating `this.state` explicitly. Use this instead: https://facebook.github.io/react/docs/update.html